tests/Microsoft.Identity.Test.LabInfrastructure/KeyVaultSecretsProvider.cs

```diff
@@ -128,30 +128,33 @@ private async Task<string> AuthenticationCallbackAsync(string authority, string
             {
                 return _authResult.AccessToken;
             }
-            var authContext = new AuthenticationContext(authority, keyVaultTokenCache);
 
             AuthenticationResult authResult;
+            IConfidentialClientApplication app;
+            X509Certificate2 cert = null;
             switch (_config.AuthType)
             {
                 case KeyVaultAuthenticationType.ClientCertificate:
-                    if (_assertionCert == null)
+                    cert = CertificateHelper.FindCertificateByThumbprint(_config.CertThumbprint);
+                    if (cert == null)
                     {
-                        var cert = CertificateHelper.FindCertificateByThumbprint(_config.CertThumbprint);
-                        if (cert == null)
-                        {
-                            throw new InvalidOperationException(
-                                "Test setup error - cannot find a certificate in the My store for KeyVault. This is available for Microsoft employees only.");
-                        }
-                        _assertionCert = new ClientAssertionCertificate(_config.ClientId, cert);
+                        throw new InvalidOperationException(
+                            "Test setup error - cannot find a certificate in the My store for KeyVault. This is available for Microsoft employees only.");
                     }
-                    authResult = await authContext.AcquireTokenAsync(resource, _assertionCert).ConfigureAwait(false);
+
+                    app = ConfidentialClientApplicationBuilder.Create(_config.ClientId)
+                                              .WithAuthority(new Uri(authority), true)
+                                              .WithCertificate(cert)
+                                              .Build();
+
+                    authResult = await app.AcquireTokenForClientAsync(new[] { resource + "/.default" }).ConfigureAwait(false);
```

Comment:
nit:  do you want to store the new[] {resource + "/.default"} into a variable since you use it in two places?